crates/subspace-core-primitives/src/lib.rs

```diff
@@ -95,23 +93,122 @@ pub type Blake3Hash = [u8; BLAKE3_HASH_SIZE];
     Hash,
     From,
     Into,
+    AsRef,
+    AsMut,
     Deref,
     DerefMut,
     Encode,
     Decode,
     TypeInfo,
     MaxEncodedLen,
 )]
+#[into(ref, ref_mut)]
+#[as_ref(forward)]
+#[as_mut(forward)]
```

Comment:
There is nothing wrong with this, but I'd honestly rather implement these manually it gets too nuanced this way. These derives are best for simple cases when we don't need to specify a bunch of additional attributes.